src/apps/bridge/bridgeLog.cpp

```diff
@@ -1,29 +1,59 @@
 #include "bridgeLog.h"
+#include "FreeRTOS.h"
 #include "app_pub_sub.h"
 #include "bm_serial.h"
 #include "device_info.h"
 #include <stdio.h>
 #include <string.h>
 
-void bridgeLogPrintf(const char *str, size_t len, bridgeLogType_e type) {
-  printf("%.*s", len, str);
-  switch (type) {
-  case BRIDGE_SYS:
-    bm_serial_pub(getNodeId(), APP_PUB_SUB_BM_BRIDGE_PRINTF_TOPIC,
-                  sizeof(APP_PUB_SUB_BM_BRIDGE_PRINTF_TOPIC) - 1,
-                  reinterpret_cast<const uint8_t *>(str), len,
-                  APP_PUB_SUB_BM_BRIDGE_PRINTF_TYPE, APP_PUB_SUB_BM_BRIDGE_PRINTF_VERSION);
-    break;
-  case BRIDGE_CFG:
-    bm_serial_pub(getNodeId(), APP_PUB_SUB_BM_BRIDGE_CFG_PRINTF_TOPIC,
-                  sizeof(APP_PUB_SUB_BM_BRIDGE_CFG_PRINTF_TOPIC) - 1,
-                  reinterpret_cast<const uint8_t *>(str), len,
-                  APP_PUB_SUB_BM_BRIDGE_CFG_PRINTF_TYPE,
-                  APP_PUB_SUB_BM_BRIDGE_CFG_PRINTF_VERSION);
-    break;
-  default:
-    printf("ERROR: Unknown log type in bridgeLogPrintf\n");
-    break;
+void bridgeLogPrint(bridgeLogType_e type, bm_common_log_level_e level, bool print_header,
+                    const char *format, ...) {
+  va_list va_args;
+  va_start(va_args, format);
+  vBridgeLogPrint(type, level, print_header, format, va_args);
+  va_end(va_args);
+}
+
+void vBridgeLogPrint(bridgeLogType_e type, bm_common_log_level_e level, bool print_header,
+                     const char *format, va_list va_args) {
+  bm_common_log_t *log_msg = NULL;
+  do {
+    int print_size = vsnprintf(nullptr, 0, format, va_args);
+    if (print_size < 0) {
+      printf("ERROR: vsnprintf failed in bridgeLogPrint\n");
+      break;
+    }
+    print_size++; // Add 1 for null terminator
+    size_t msg_len = sizeof(bm_common_log_t) + print_size;
+    log_msg = static_cast<bm_common_log_t *>(pvPortMalloc(msg_len));
+    configASSERT(log_msg);
+    memset(log_msg, 0, msg_len);
+    log_msg->level = level;
+    log_msg->message_length = print_size;
+    log_msg->print_header = print_header;
+    vsnprintf(log_msg->message, print_size, format, va_args);
```

Comment:
I see that the `timestamp_utc_s` field isn't set. Is that intentional, or should we set it?

Reply:
Good catch we should set it! (we currently don't use it at the moment, but we may want to later)